### DISCUSSION TITLE
How to toggle NerdTree panel

### DISCUSSION THREAD

**Discussion by ChrisTaylorDeveloper** [JetBrains/ideavim#747] [Q&A] [answered]
How can I toggle open and closed the NerdTree file panel?

I have tried several different mappings in .ideavimrc but none of them work 100%.  The key combination I want to use is Leader followed by the letter e.  This key combination should both open and close the panel.

Here is my latest attempt
```
function ToggleNERDTree()
  normal! <Esc>
  :NERDTreeToggle<CR>
endfunction
map <leader>e :call ToggleNERDTree()<CR>
```
If the panel is closed and I press Leader e, the panel opens as expected.  However, when I press Leader e again, to close the panel, the e character is captured by the file-tree and a search for the letter 'e' is started.

How can I achieve my objective?  Thank you.

**Comment by citizenmatt** [comment, marked as answer]
Unfortunately, this isn't currently possible, because IdeaVim is an editor extension, so IdeaVim mappings can only work in the context of an editor. Once the Project View has focus, you're no longer in an editor context, and standard mappings won't work.

You can use normal IDE shortcuts, such as `Escape` to move the focus back to an editor, or `Shift+Escape` to close the current tool window. You can also use the Project View shortcut such as `Alt/Cmd+0` to toggle it. There might also be a NERDTree shortcut, but I'm not familiar with it and would have to check.

**Reply by ChrisTaylorDeveloper** [reply]
Thank you.

**Reply by SerkanSipahi** [reply]
@citizenmatt I have the same problem. I can not toggle the NerdTree when using `:NERDTreeToggle`. The only working way is using this shortcut `Alt/Cmd+0`. 

Is it possible to trigger these shortcut? I tried these mappings but no one of them is working
```bash
nnoremap <D-\> <D-1>
map <D-\> <D-1>
```

**Reply by citizenmatt** [reply]
I'm not sure what those mappings are intended to do, but as mentioned above, IdeaVim is an editor extension, so IdeaVim mappings will only work while there's editor has focus. Once the Project View has focus, normal IdeaVim mappings don't get processed.

**Reply by SerkanSipahi** [reply]
@citizenmatt 

> I'm not sure what those mappings are intended to do

When pressing `cmd+backslash` it should trigger the following shortcut `cmd+1`. When this could work, then the tree toggle works.

**Reply by SerkanSipahi** [reply]
> Once the Project View has focus, normal IdeaVim mappings don't get processed.

I understood, thanks.

**Reply by citizenmatt** [reply]
Ah. Unfortunately, it's not possible to map _to_ and IDE shortcut. They get processed differently to how IdeaVim keys are processed.

The best you could do is map to a specific IDE action, something like:

```
nmap <D-\> <Action>(ActivateProjectToolWindow)
```

This will work to activate the Project window, but once active, the editor is no longer able to process IdeaVim mappings and it won't toggle. In this case, the default IDE `<Esc>` shortcut will move the focus from the tool window back to the editor.
